build/gen_stub.php

```diff
@@ -1810,7 +1810,9 @@ function handleStatements(FileInfo $fileInfo, array $stmts, PrettyPrinterAbstrac
 }
 
 function parseStubFile(string $code): FileInfo {
-    $lexer = new PhpParser\Lexer();
+    $minCompatiblePhpVersionId = 80100; // PHP 8.1
+
+    $lexer = PHP_VERSION_ID >= $minCompatiblePhpVersionId ? new PhpParser\Lexer() : new PhpParser\Lexer\Emulative();
```

Comment:
You can always use the emulative lexer, it just won't do anything for older versions.

Reply:
I thought that it's slower than the native one, that's why I added this condition. But I guess the speed difference is negligible, at worst :)